app/models/list.rb

```diff
@@ -125,4 +125,16 @@ def clear_cache(host = nil)
   def self.default_network
     find(Lilsis::Application.config.default_network_id)
   end
+
+  # [Entity|Ids]
+  def add_entities(entities_or_ids)
+    entities_or_ids.each{ |ei| add_entity(ei) }
```

Comment:
by calling add_entity in a loop we are executing 'touch' on the list after every entity, which could amount to a lot of surpurflous sql calls. we should call 'touch' once at the end of the batch upload.

Reply:
+1

Reply:
(nice catch!)